Add parameter to ModalResult that identifies the modal which caused the close event

larsk2009 opened the pull request:
This commit could potentially fix issue #76. I don't know if you think this is a good solution, but than at least we have a starting point. I have only made the change and not made any examples for it yet. I can do this in the future, but for now I would like to see if you like this change I made.

The type of the modal that was used to show the modal will be returned in the ModalResult of the OnClose event. This way a page that has multiple components could see which component was just closed.

Again, I am not saying this is a perfect solution, just something I thought of that could be a good idea for issue #76. Just let me know what you think.

chrissainty commented on src/Blazored.Modal/Services/ModalService.cs:
```diff
@@ -20,6 +20,8 @@ public class ModalService : IModalService
         /// </summary>
         internal event Action<string, RenderFragment, ModalParameters, ModalOptions> OnShow;
 
+        private Type ModalType;
```
```suggestion
        private Type _modalType;
```

chrissainty commented on src/Blazored.Modal/Services/ModalService.cs:
```diff
@@ -68,6 +70,7 @@ public class ModalService : IModalService
             }
 
             var content = new RenderFragment(x => { x.OpenComponent(1, typeof(T)); x.CloseComponent(); });
+            ModalType = typeof(T);
```
```suggestion
            _modalType = typeof(T);
```

chrissainty commented on src/Blazored.Modal/Services/ModalService.cs:
```diff
@@ -78,7 +81,7 @@ public class ModalService : IModalService
         {
             CloseModal?.Invoke();
-            OnClose?.Invoke(ModalResult.Cancel());
+            OnClose?.Invoke(ModalResult.Cancel(ModalType));
```
```suggestion
            OnClose?.Invoke(ModalResult.Cancel(_modalType));
```

chrissainty commented on src/Blazored.Modal/Services/ModalService.cs:
```diff
@@ -87,6 +90,7 @@ public void Cancel()
         /// <param name="modalResult"></param>
         public void Close(ModalResult modalResult)
         {
+            modalResult.ModalType = ModalType;
```
```suggestion
            modalResult.ModalType = _modalType;
```

larsk2009 commented:
Thank you for your feedback @chrissainty.

I fixed the naming convention mistake. 

For your question. You did understand it correctly. My use case was that one page has multiple modals, but they will only be opened one at a time. (Think a delete modal and an add modal, where you want to do something different depending on which modal was closed). In this use case my implementation would work I think, but you are right that it would not work with multiple modals at the same time. 

Now that I am thinking about it, can you even have multiple modals open at the same time? Wouldn't a new modal just 'override' the last one, or am I missing something here?

chrissainty commented:
Multiple modals is a request I've had for a while now, but I've always been conflicted about the best way to implement them. 

Thank you for fixing the naming issue. Would you be able to add an example of using this into the sample project and the readme as well? I think I'm happy to go ahead with your proposed solutions.

larsk2009 commented:
I added the example and updated the readme. I had some trouble coming up with a good name for the example so see if what I decided on makes sense.

I'll admit I am not very good at writing documentation such as the readme, and I am trying to improve, so let me know if something is not clear!

larsk2009 commented:
No problem, glad I could add something usefull!
